order/src/main/java/com/caldevsupplychain/order/controller/OrderControllerImpl.java

```diff
@@ -43,25 +48,36 @@
 
 	private ApiErrorsExceptionHandler apiErrorsExceptionHandler;
 
-	@PostMapping("/order")
+	@PostMapping
 	@RequiresJwtAuthentication
 	@RequiresPermissions("order:create")
 	public ResponseEntity<?> createOrder(@Validated @RequestBody OrderWS orderWS) {
 
-		// check user_id
-		if(!accountService.findById(orderWS.getUserId()).isPresent()){
-			log.error("Error in create order. User id={} not found", orderWS.getUserId().toString());
-			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.USER_ID_NOT_FOUND.name(), "In create order user id={}" + orderWS.getUserId().toString().toString() + " not found"), HttpStatus.NOT_FOUND);
-		}
+		ResponseEntity<?> validateUserAgentUuidResult = validateUserAndAgentUuid(orderWS);
+
 
-		// check agent_id
-		if(!accountService.findById(orderWS.getAgentId()).isPresent()){
-			log.error("Error in create order. Agent id={} not found", orderWS.getAgentId().toString());
-			return new ResponseEntity<>(new ApiErrorsWS(ErrorCode.AGENT_ID_NOT_FOUND.name(), "In create order agent id={}" + orderWS.getAgentId().toString().toString() + " not found"), HttpStatus.NOT_FOUND);
+		if(validateUserAgentUuidResult != null) {
+			return validateUserAgentUuidResult;
 		}
 
+		String currentUserUuid = (String) SecurityUtils.getSubject().getPrincipal();
```

Comment:
We should prob have a util class for this since current user context is used pretty frequently
```
@Component
public class ContextUtils {
        Optional<UserBean> currentUser() {
                .....
        }
}

```

Reply:
Fixed